dashboard/app/controllers/lessons_controller.rb

```diff
@@ -5,30 +5,14 @@ class LessonsController < ApplicationController
 
   # GET /lessons/1
   def show
-    @lesson_data = {
-      title: @lesson.localized_title,
-      overview: @lesson.overview,
-      announcements: @lesson.announcements,
-      purpose: @lesson.purpose,
-      preparation: @lesson.preparation
-    }
+    @lesson_data = @lesson.to_camelized_hash
+    @lesson_data[:title] = @lesson.localized_title
   end
 
   # GET /lessons/1/edit
   def edit
     @lesson_data = {
-      editableData: {
-        name: @lesson.name,
-        overview: @lesson.overview,
-        studentOverview: @lesson.student_overview,
-        assessment: @lesson.assessment,
-        unplugged: @lesson.unplugged,
-        lockable: @lesson.lockable,
-        creativeCommonsLicense: @lesson.creative_commons_license,
-        purpose: @lesson.purpose,
-        preparation: @lesson.preparation,
-        announcements: @lesson.announcements
-      }
+      editableData: @lesson.to_camelized_hash
```

Comment:
Problem 2: if i try to fix Problem 1 by moving "title" into the serializer, then I would want to nest the data returned by the serializer like this:
```
{
  title: ...
  editableData: {...}
}
```
but that format doesn't really make sense for the `show` action, because it doesn't have anything to do with editing.

OTOH if I use `lesson` instead of `editableData`, then I can have the serializer return:
```
```
{
  title: ...
  lesson: {...}
}
```
but as you pointed out earlier @uponthesun this is also confusing because the things outside of `lesson` still have to do with "lesson".

a third option might be to make the serializer conditionally nest things under `editableData` or not, depending on an parameter you pass into it via https://github.com/rails-api/active_model_serializers/blob/v0.10.6/docs/howto/passing_arbitrary_options.md . but at that point, the output of the serializer might be so different for edit / update that it could be simpler to just write two separate serializers, which seems like it might be a step in the wrong direction.

Any thoughts, @uponthesun ?

Reply:
I think I like having two separate serializers due to simplicity, as long as they're not duplicating fields / able to get out of sync. (Unless we explicitly want to decouple them, but it seems like right now we don't.)

Serializers feel lightweight enough such that it's fine to have multiples for the same class, as long as they're for specific use cases, and named clearly. I think we follow this pattern already with some of the PLC serializers.

Some possible implementations:
* One serializer for only the editable fields, and one for the noneditable fields.
* One serializer for editable, and one for everything, which references the attributes list of the editable one.

One more thing - let's adopt the mindset of trusting our unit tests for these APIs, and changing these kinds of implementation details later without worrying, if we find something better later. (Not realistic for everything, of course, but I think these APIs should be simple enough to be comprehensively covered by unit tests.)